### DISCUSSION TITLE
Styling Individual Nodes of a DirectoryTree

### DISCUSSION THREAD

**Discussion by b-peri** [Textualize/textual#3802] [Q&A] [answered]
Hi all! I'm currently working on a Textual app in which I'm hoping to display the "Upload" status of individual files in a user's local directory using a dynamically-styled DirectoryTree widget (see image below). My initial idea here was to somehow "select" individual nodes in my DirectoryTree based on their path names, and then apply additional styling using e.g. CSS classes. However, after having a reasonably thorough read through the docs, I'm no longer sure it will be so straightforward.

![image](https://github.com/Textualize/textual/assets/77279592/b556256f-89c8-4ec6-9f6b-17775bbf33d1)

Is it all possible to select and style individual nodes of a DirectoryTree in this way? If not, is there perhaps some other workaround I could consider using? I apologize if this is an obvious question - I'm relatively new to Textual and to software development in Python in general, so maybe I'm just missing something here!

**Comment by davep** [comment, marked as answer]
Nodes in a `Tree` aren't widgets, and so aren't part of the DOM, and as such can't be styled (directly) with CSS. Likely the best approach if you want custom highlighting is to inherit from `DirectoryTree` and extend `render_label`. See [`DirectoryTree.render_label`](https://github.com/Textualize/textual/blob/dc43f4447fc5e1a72bb9648c1e4c619875e48395/src/textual/widgets/_directory_tree.py#L282-L324) to see how it does the current styling.

**Reply by b-peri** [reply]
Thanks for the very prompt response @davep! I'll have a look and let you know if I run into any further issues!
